test/passthrough_test.js

```diff
@@ -38,3 +38,84 @@ asyncTest('allows matched paths to be pass-through', function(assert) {
     }
   });
 });
+
+asyncTest('asynchronous request with pass-through has timeout, withCredentials and onprogress event', function(assert) {
+  function testXHR() {
+    this.pretender = pretender;
+    this.open = function() {};
+    this.setRequestHeader = function() {};
+    this.send = {
+      apply: function(xhr, argument) {
+        assert.equal(xhr.timeout, 1000);
+        assert.equal(xhr.withCredentials, true);
+        assert.ok(xhr.onprogress !== undefined);
+        this.pretender.resolve(xhr);
+      }
+    };
+  }
+  pretender._nativeXMLHttpRequest = testXHR;
+
+  pretender.post('/some/path', pretender.passthrough);
+
+  $.ajax({
```

Comment:
Its not really important, I just followed the other test since majority was using $.ajax.